packages/node-core/src/indexer/dictionary/dictionary.service.ts

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020-2023 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import {EventEmitter2} from '@nestjs/event-emitter';
+import {NETWORK_FAMILY} from '@subql/common';
+import fetch from 'cross-fetch';
+import {NodeConfig} from '../../configure';
+import {getLogger} from '../../logger';
+import {BlockHeightMap} from '../../utils/blockHeightMap';
+import {DictionaryResponse, DictionaryVersion, IDictionary, IDictionaryCtrl} from './types';
+import {subqlFilterBlocksCapabilities} from './utils';
+import {DictionaryV2Metadata} from './';
+
+export async function inspectDictionaryVersion(endpoint: string): Promise<DictionaryVersion> {
+  if (endpoint.includes('/rpc')) {
```

Comment:
I don't think you can assume the url includes `/rpc`

Reply:
okay, now if `subqlFilterBlocksCapabilities` throw an error we will always use v1